include/seqan3/range/view/pairwise_combine.hpp

```diff
@@ -0,0 +1,597 @@
+     * \{
+     */
+    constexpr pairwise_combine_view() = default;
+    constexpr pairwise_combine_view(pairwise_combine_view const &) = default;
+    constexpr pairwise_combine_view(pairwise_combine_view &&) = default;
+    constexpr pairwise_combine_view & operator=(pairwise_combine_view const &) = default;
+    constexpr pairwise_combine_view & operator=(pairwise_combine_view &&) = default;
+    ~pairwise_combine_view() = default;
+
+    /*!\brief Constructs from a view.
+     * \param[in] range A rvalue reference to the source view.
+     *
+     * \details
+     *
+     * During construction the iterator pointing to the last element of the view is cached. This optimises
+     * the call to end if the source range models only ForwardRange. Otherwise the call to end will be
+     * linear in the number of elements of the source range.
+     */
+    explicit constexpr pairwise_combine_view(source_range_type && range) : src_range{std::move(range)}
+    {
+        // Check if range is empty.
+        if (std::ranges::begin(src_range) == std::ranges::end(src_range))
+        {
+            back_iterator = std::ranges::end(src_range);
+        }
+        else
+        {
+            if constexpr (std::ranges::BidirectionalRange<source_range_type> &&
+                          std::ranges::CommonRange<source_range_type>)
+            {
+                back_iterator = std::ranges::prev(std::ranges::end(src_range));
+            }
+            else
+            {  // Cannot decrement so we invoke linear pass until we get to the last element.
+                size_t size = std::distance(src_range);
+                back_iterator = std::ranges::begin(src_range);
+                std::advance(back_iterator, size - 1);
+            }
+        }
+    }
+
+    /*!\brief Constructs from a range.
+     * \param[in] range A lvalue reference to the source range.
+     *
+     * \details
+     *
+     * Wraps the passed range in a ref_view in order to store lvalue references to ranges (non-views) as a view
+     * and delegate to the view constructor above.
+     */
+    template <typename other_range_type>
+    explicit constexpr pairwise_combine_view(other_range_type & range) : pairwise_combine_view{view::all(range)}
+    {}
+    //!\}
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the first element of the range.
+     * \returns Iterator to the first element.
+     *
+     * If the range is empty, the returned iterator will be equal to end().
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr iterator_type begin() noexcept
+    {
+        return iterator_type{std::ranges::begin(src_range), std::ranges::begin(src_range), std::ranges::end(src_range)};
+    }
+
+    //!\copydoc begin()
+    constexpr iterator_type begin() const noexcept
+        requires const_iterable_concept<source_range_type>
```

Comment:
did you check if doxygen ignores the requires? sometimes it does, sometimes not..

Reply:
I removed it completely, since the view does not propagate const to the source range.